### PR TITLE
Added aframe lightmap controls to io3d-data3d and io3d-furniture

### PR DESCRIPTION
**Pull request by Aadjou**
Scope:
The feature implements a-frame controls for lightMap intensity & center for io3d-data3d and io3d-furniture.

Design:
- Based on the discussions in this [issue](https://github.com/archilogic-com/3dio-js/issues/13)

How to use:
- The lightMap controls can be set directly in the a-frame entity
`<a-entity io3d-data3d="key:/lighting.gz.data3d.buffer; lightMapIntensity:0.1; lightMapCenter:0.5"></a-entity>`
`  <a-entity io3d-furniture="id:10a54bcf-3b9c-4518-b7ea-81c4251cf5a4;  lightMapIntensity:0.1; lightMapCenter:0.5" ></a-entity>`
- The settings are optional, if they are not set, the lightMapSetting of the data3d file will be used
- If both are not set, fallback values will be used (defined in _set-material.js_)

- additionally, the values can be set/manipulated in the a-frame scene inspector

Note:
- The controls will not influence materials that have no lightmap assigned. Therefore only lightmap materials will adapt
- Therefore most furniture will not adjust, since they don't have a lightmap channel.

### CODE REVIEW

File: src/aframe/three/data3d-view/set-material.js [hunk 1]
```diff
@@ -135,8 +137,23 @@ export default function setMaterial (args) {
 
   // lightmap settings
   if (_attributes.mapLight || _attributes.mapLightPreview) {
-    material3d.lightMapIntensity = (_attributes.mapLightIntensity !== undefined) ? _attributes.mapLightIntensity : 1.0
-    material3d.lightMapCenter = (_attributes.mapLightCenter !== undefined) ? _attributes.mapLightCenter : 0.5
+    var lmi = 1.0
+    var lmc = 0.5
```
Comment (frederic-schwarz) [inline src/aframe/three/data3d-view/set-material.js]: @Aadjou I assume these are the defaults.
I think we should add

**Comment by frederic-schwarz** [comment]
my comments are in the issue https://github.com/archilogic-com/3dio-js/issues/13#issuecomment-316826530

File: src/aframe/three/data3d-view/set-material.js (hunk 1 above)
Comment (Aadjou) [inline src/aframe/three/data3d-view/set-material.js]: will do

**Comment by tomas-polach** [comment]
great job with the lightmaps!!
i updated component schemas following aframe specs in https://github.com/archilogic-com/3dio-js/commit/2b6a752539e1efbc4a6dfb30613d11779eaff753 maybe @AVGP can doublecheck?

while testing i figured out that changing lightMapCenter rather seems to change lightmap falloff - is it possible that there is a glitch in how the variables are wired? :)

lightMapCenter:1
localhost_8087_examples-browser_aframe__and_professional-model-indoor-view_jpg___117___rgb_8__

lightMapCenter:0.1 (lightmap should be darker overall)
localhost_8087_examples-browser_aframe_

lightMapCenter:2 (dark artefacts on bright places as it is common in high falloff settings)
localhost_8087_examples-browser_aframe_ 2

**Comment by Aadjou** [comment]
thanks for all the feedback. very helpful!
that could be a glitch. i will check it out
please check my comment above on commit 2b6a752 
i was using custom property schema so we can have a fallback to the values of the buffer file
 i think that commit should be reverted @avgp @tomas

**Comment by Aadjou** [comment]
open issues:
- Why do the controls behave so differently from the editor
- is there a glitch with exposure and falloff?
- negative values on exposure? 
- do the fallbacks work properly?
- should lightMapCenter also be renamed in Shader & material3d?
- 0.0 should be treated as 0.0 and not as undefined hence fallback to default
- lightMapExposure seems to behave different than Exposure settings in the Editor -> way softer

resolved:
- renamd lightMapCenter to lightMapExposure 
- documentation of fallback values

**Comment by tomas-polach** [comment]
i will change lightmap falloff default to 0 (= turned off) because the current default 0.5 easily produces irrational artefacts (bright areas turning black):

**Comment by tomas-polach** [comment]
yes pls. suggesting solutions is always welcome :)

**Comment by Aadjou** [comment]
LMI: 1.0
LME: 0.0
1_0and0_0
LMI: 1.0
LME: 0.5
1_0and0_5
LMI: 2.0
LME: 0.5
2_0and0_5

no lights in the scene

**Comment by Aadjou** [comment]
I am ready for a second round of reviews

**Pull request merged**